tests/integ/test_dataframe.py

```diff
@@ -2620,7 +2621,20 @@ def test_save_as_table_respects_schema(session, save_mode, local_testing_mode):
             )
             with pytest.raises(SnowparkSQLException, match=exception_msg):
                 df3.write.save_as_table(table_name, mode=save_mode)
-        else:  # save_mode in ('append', 'errorifexists')
+        elif save_mode == "append":
+            # As long as the dataframe that is appended is only missing columns
+            # that are nullable there will be no error raised.
```

Comment:
I don't think this is the behavior in snowflake? -- in live tests, this triggers, see CI: https://github.com/snowflakedb/snowpark-python/actions/runs/9846922564/job/27185862288?pr=1872

`E   snowflake.snowpark.exceptions.SnowparkSQLException: (1304): 01b589f8-090a-a2cf-000c-a904df8facdf: 002020 (21S01): SQL compilation error:
E   Insert value list does not match column list expecting 2 but got 1`